src/test/java/io/strimzi/kafka/bridge/http/HttpCorsTests.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+package io.strimzi.kafka.bridge.http;
+
+import io.strimzi.kafka.bridge.HealthChecker;
+import io.strimzi.kafka.bridge.config.BridgeConfig;
+import io.strimzi.kafka.bridge.facades.KafkaFacade;
+import io.strimzi.kafka.bridge.utils.Urls;
+import io.vertx.core.Vertx;
+import io.vertx.core.VertxOptions;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.ext.web.client.WebClientOptions;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.hasItem;
+
+@ExtendWith(VertxExtension.class)
+@SuppressWarnings({"checkstyle:JavaNCSS"})
+public class HttpCorsTests {
+
+    static final Logger LOGGER = LogManager.getLogger(HttpCorsTests.class);
+    static Map<String, Object> config = new HashMap<>();
+    static long timeout = 5L;
+
+    static {
+        config.put(HttpConfig.HTTP_CONSUMER_TIMEOUT, timeout);
+        config.put(BridgeConfig.BRIDGE_ID, "my-bridge");
+    }
+
+    static Vertx vertx;
+    static HttpBridge httpBridge;
+    static WebClient client;
+
+    static BridgeConfig bridgeConfig;
+    static KafkaFacade kafkaCluster = new KafkaFacade();
+
+    @BeforeAll
+    static void beforeAll() {
+        kafkaCluster.start();
+    }
+
+    @BeforeEach
+    void prepare() {
+        VertxOptions options = new VertxOptions();
+        options.setMaxEventLoopExecuteTime(Long.MAX_VALUE);
+        vertx = Vertx.vertx(options);
+    }
+
+    @AfterEach
+    void cleanup() {
+        vertx.close();
+    }
+
+    @AfterAll
+    static void afterAll() {
+        kafkaCluster.stop();
+    }
+
+    @Test
+    public void testCorsNotEnabled(VertxTestContext context) {
+        createWebClient(false, null, context);
+
+        if (!"TRUE".equalsIgnoreCase(System.getenv("STRIMZI_USE_SYSTEM_BRIDGE"))) {
+            vertx.deployVerticle(httpBridge, context.succeeding(id -> client
+                .request(HttpMethod.OPTIONS, 8080, "localhost", "/ready")
+                .putHeader("Origin", "https://evil.io")
+                .putHeader("Access-Control-Request-Method", "GET")
```

Comment:
so you are simulating a "preflight" request where the client "is asking for information" if the server is willing to accept GET on the specified origin. Should you even test without a "preflight" request when the CORS is disabled as well? So having the client just trying a GET and see what happens?

Reply:
yes it makes sense, I will split into 2 methods:
testCorsNotEnabled
testPreflightCorsNotEnabled